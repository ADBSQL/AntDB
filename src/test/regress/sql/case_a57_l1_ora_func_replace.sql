set grammar to oracle;
select replace('abc', 'a', 'A');
select replace('abc', 'a', '');
select replace('abc', 'a', NULL);
select replace('abc', 'a');
select replace('abc', '', 'A');
select replace('abc', NULL, 'A');
select replace('abc', NULL);
select replace('', 'a', 'A');
select replace(NULL, 'a', 'A');
select replace(NULL, '', 'A');
select replace(NULL, NULL, 'A');
select replace(NULL, 'a', '');
select replace(NULL, 'a', NULL);
select replace(NULL, '', '');
select replace(NULL, '', NULL);
select replace(NULL, NULL, '');
select replace(NULL, NULL, NULL);
select replace('', '', '');
select replace(123, 2, 3);
select replace(123, 2);
select replace('我是，中国人；','，',',');


create table t4test(id int, txt varchar);
insert into t4test values(1, 'zhang****liang');
insert into t4test values(2, 'zhangying***');
insert into t4test values(3, replace('1234567','45','$$'));
select * from t4test order by id;
update t4test set txt=replace(txt,'*','@');
select * from t4test order by id;
drop table t4test;
