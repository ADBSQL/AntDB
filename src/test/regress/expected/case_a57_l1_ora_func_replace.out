set grammar to oracle;
select replace('abc', 'a', 'A');
 REPLACE 
---------
 Abc
(1 row)

select replace('abc', 'a', '');
 REPLACE 
---------
 bc
(1 row)

select replace('abc', 'a', NULL);
 REPLACE 
---------
 bc
(1 row)

select replace('abc', 'a');
 REPLACE 
---------
 bc
(1 row)

select replace('abc', '', 'A');
 REPLACE 
---------
 abc
(1 row)

select replace('abc', NULL, 'A');
 REPLACE 
---------
 abc
(1 row)

select replace('abc', NULL);
 REPLACE 
---------
 abc
(1 row)

select replace('', 'a', 'A');
 REPLACE 
---------
 
(1 row)

select replace(NULL, 'a', 'A');
 REPLACE 
---------
 
(1 row)

select replace(NULL, '', 'A');
 REPLACE 
---------
 
(1 row)

select replace(NULL, NULL, 'A');
 REPLACE 
---------
 
(1 row)

select replace(NULL, 'a', '');
 REPLACE 
---------
 
(1 row)

select replace(NULL, 'a', NULL);
 REPLACE 
---------
 
(1 row)

select replace(NULL, '', '');
 REPLACE 
---------
 
(1 row)

select replace(NULL, '', NULL);
 REPLACE 
---------
 
(1 row)

select replace(NULL, NULL, '');
 REPLACE 
---------
 
(1 row)

select replace(NULL, NULL, NULL);
 REPLACE 
---------
 
(1 row)

select replace('', '', '');
 REPLACE 
---------
 
(1 row)

select replace(123, 2, 3);
 REPLACE 
---------
 133
(1 row)

select replace(123, 2);
 REPLACE 
---------
 13
(1 row)

select replace('我是，中国人；','，',',');
    REPLACE    
---------------
 我是,中国人；
(1 row)

create table t4test(id int, txt varchar);
insert into t4test values(1, 'zhang****liang');
insert into t4test values(2, 'zhangying***');
insert into t4test values(3, replace('1234567','45','$$'));
select * from t4test order by id;
 ID |      TXT       
----+----------------
  1 | zhang****liang
  2 | zhangying***
  3 | 123$$67
(3 rows)

update t4test set txt=replace(txt,'*','@');
select * from t4test order by id;
 ID |      TXT       
----+----------------
  1 | zhang@@@@liang
  2 | zhangying@@@
  3 | 123$$67
(3 rows)

drop table t4test;
